docs/en/02_Developer_Guides/08_Performance/02_HTTP_Cache_Headers.md

```diff
@@ -1,7 +1,125 @@
 title: HTTP Cache Headers
 summary: Set the correct HTTP cache headers for your responses.
 
-# Caching Headers
+# HTTP Cache Headers
+
+## Overview
+
+By default, SilverStripe sends headers which signal to HTTP caches
+that the response should be considered not cacheable.
+HTTP caches can either be intermediary caches (e.g. CDNs and proxies), or clients (e.g. browsers).
+The cache headers sent are `Cache-Control: no-store, no-cache, must-revalidate`;
+
+HTTP caching can be a great way to speed up your website, but needs to be properly applied.
+Getting it wrong can accidentally expose draft pages or other protected content.
+The [Google Web Fundamentals](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching#public_vs_private)
+are a great way to learn about HTTP caching.
+
+## Cache Control Headers
+
+In order to support developers in making safe choices around HTTP caching,
+we're using a `HTTPCacheControl` class to control if a response
+should be considered public or private. This is an abstraction on existing
+lowlevel APIs like `HTTP::add_cache_headers()` and `SS_HTTPResponse->addHeader()`.
+
+The `HTTPCacheControl` API makes it easier to express your caching preferences
+without running the risk of overriding essential core safety measures.
+Most commonly, these APIs will prevent HTTP caching of draft content.
+
+It will also prevent content generated with an active session,
+since the system can't tell whether session data was used to vary the output.
+In this case, it's up to the developer to opt-in to caching,
+after ensuring that certain execution paths are safe despite of using sessions.
+
+The system behaviour does not guard against accidentally caching "private" content,
+since there are too many variations under which output could be considered private
+(e.g. a custom "approval" flag on a comment object). It is up to
+the developer to ensure caching is used appropriately there.
+
+The `HTTPCacheControl` class supplements the `HTTP` helper class.
+It comes with methods which let developers safely interact with the `Cache-Control` header.
+
+ * `disableCache()`: Disables caching in intermediary caches as well as HTTP clients.
+    Removes all state and replaces it with `no-cache, no-store, must-revalidate`.
+ * `privateCache()`: Disables caching in intermediary caches, but allows it in HTTP clients.
+    Sends `Cache-Control: private`.
```

Comment:
do we want to make it clear that `privateCache` and `publicCache` just add the `public` and `private` directive rather than replacing the whole set of directives with `public` and `private`?

Reply:
I'm adding that to the PHPDoc, and copying those descriptions into the upgrade guide and HTTP caching docs. Easiest to keep wording consistent to avoid confusion there.